### PR TITLE
Version Component - Replaced classes by func. comp

### PR DESCRIPTION
**Pull request by priscilawebdev**
**Type:** Refactor

**The following has been addressed in the PR:** https://github.com/verdaccio/ui/issues/116

**Unit or Functional tests are included in the PR?**

It was not necessary to update the existing ones

**Description:**

in order to use react hooks, in this PR I updated the Version component by changing it from a class to a functional component. I have also splitted it's content so that it is clearer.

### CODE REVIEW

File: src/components/Versions/Versions.tsx [hunk 1]
```diff
@@ -1,90 +1,48 @@
...
 
-import { Heading, Spacer, ListItemText } from './styles';
+import { Heading } from './styles';
+
+import VersionsTagList from './VersionsTagList';
+import VersionsHistoryList from './VersionsHistoryList';
+
+import { DIST_TAGS } from '../../../lib/constants';
 
 export const NOT_AVAILABLE = 'Not available';
```
Comment (juanpicado) [inline src/components/Versions/Versions.tsx]: Not longer use, we can remove it.

File: src/components/Versions/Versions.tsx [hunk 2]
```diff
@@ -1,90 +1,48 @@
...
-            <Heading variant="subtitle1">{LABEL_VERSION_HISTORY}</Heading>
-            {this.renderPackageList(versions, timeMap, packageName)}
-          </>
-        )}
-      </>
-    );
-  }
-}
+const Versions: React.FC = () => (
+  <DetailContextConsumer>
```
Comment (juanpicado) [inline src/components/Versions/Versions.tsx]: I think you can use here https://reactjs.org/docs/hooks-reference.html#usecontext

The code will be even smaller and simpler to read.

File: src/components/Versions/styles.ts [hunk 3]
```diff
@@ -14,6 +14,7 @@ export const Spacer = styled('div')({
   borderBottom: '1px dotted rgba(0, 0, 0, 0.2)',
   whiteSpace: 'nowrap',
   height: '0.5em',
+  margin: '0 16px',
```
Comment (juanpicado) [inline src/components/Versions/styles.ts]: Could you give more context about the margin?

Comment (priscilawebdev) [reply src/components/Versions/styles.ts]: I noticed that there was no space between the Tabs component and the content, so I added 16px margin. Maybe in the future we may have some common spaces, making the application more homogeneous.

File: src/components/Versions/Versions.tsx (hunk 2 above)
Comment (priscilawebdev) [reply src/components/Versions/Versions.tsx]: oh true :-) good catch!

File: src/components/Versions/Versions.test.tsx [hunk 4]
```diff
@@ -1,48 +1,39 @@
...
-  beforeEach(() => {
-    jest.resetModules();
-  });
-
   afterEach(() => {
     cleanup();
   });
 
   // FIXME: this test is not deterministic (writes `N days ago` in the snapshot, where N is random number)
   test.skip('should render the component in default state', () => {
```
Comment (juanpicado) [inline src/components/Versions/Versions.test.tsx]: Let's remove this in another PR.

**Pull request merged**